rfc/0054-sdk3-management-apis.md

```diff
@@ -592,6 +599,325 @@ Nothing
 
 * Any exceptions raised by the underlying platform
 
+# CollectionQueryIndexManager
+
+This interface contains the means for managing collection-level indexes used for queries.
+
+It's a cleaner solution for this than the scopeName and collectionName parameters added to QueryIndexManager option blocks.
+Those should now be deprecated.
+
+The interface is identical to QueryIndexManager, except with the removal of the `string bucketName` parameter.
+
+CollectionQueryIndexManager appears on the Collection interface, in the form `collection.queryIndexes()`.
+
+All queries will be sent with a `query_context` parameter of "default:`bucket`.`scope`".  
+This is a mandatory parameter for 7.5, and a primary driver for adding this API.
+(It also means that this API cannot be used with servers below 7.0.)
+
+```
+public interface CollectionQueryIndexManager {
+    Iterable<QueryIndex> GetAllIndexes(GetAllQueryIndexOptions options);
```

Comment:
GetAllQueryIndexIOptions still contains the ScopeName and CollectionName fields, since they were optional at the Cluster level. I can obsolete them and throw exceptions if they are used at this level, but this is a bit awkward.

Reply:
Yea that's correct. It was this or create new options blocks with awkward names. Neither is a perfect solution, and we went with this approach when David raised it in the last meeting.

I've called out here that those parameters need to be deprecated. I haven't called out that an InvalidArgumentException should be raised if they're used with CollectionQueryIndexManager - guess I'd better do another PR.